apps/sensenet/src/components/appbar/desktop-nav-menu.tsx

```diff
@@ -65,44 +92,66 @@ export const DesktopNavMenu: React.FunctionComponent = () => {
   const globalClasses = useGlobalStyles()
   const theme = useTheme()
   const service = injector.getInstance(PersonalSettings)
-  const { openDialog } = useDialog()
-  const [open, setOpen] = useState(false)
-  const anchorRef = useRef<HTMLButtonElement>(null)
   const currentUser = useCurrentUser()
   const repo = useRepository()
   const localization = useLocalization()
+  const { openDialog } = useDialog()
+  const [openLogoutMenu, setOpenLogoutMenu] = useState(false)
+  const [openHelpMenu, setOpenHelpMenu] = useState(false)
+  const logoutRef = useRef<HTMLButtonElement>(null)
+  const helpRef = useRef<HTMLButtonElement>(null)
 
-  const handleToggle = () => {
-    setOpen((prevOpen) => !prevOpen)
+  const handleToggle = (setter: React.Dispatch<React.SetStateAction<boolean>>) => {
+    setter((prevState) => !prevState)
   }
 
-  const handleClose = (event: React.MouseEvent<EventTarget>) => {
-    if (anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) {
+  const handleClose = (
+    event: React.MouseEvent<EventTarget>,
+    ref: React.RefObject<HTMLButtonElement>,
+    setter: React.Dispatch<React.SetStateAction<boolean>>,
+  ) => {
+    if (ref.current && ref.current.contains(event.target as HTMLElement)) {
```

Comment:
ref.current?.

Reply:
what does it do?

Reply:
https://hu.reactjs.org/docs/hooks-reference.html#useref
useRef returns a mutable ref object whose .current property is initialized to the passed argument (initialValue). The returned object will persist for the full lifetime of the component.